### PR TITLE
[NFS-Ganesha][Phase2]Add testcase automation

### PR DESCRIPTION
**Pull request by MohitBis**
[NFS-Ganesha][Phase2]Add testcase automation
Scenario: Hard links : Create files from C1 and create synlinks from C2 and perform lookups from C3

### CODE REVIEW

**Comment by MohitBis** [comment]
Log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-CCACYI/

**Review by msainiRedhat — APPROVED** [review]
LGTM

**Review by pranavprakash20 — DISMISSED** [review]
LGTM

File: tests/nfs/nfs_verify_file_ops_hard_links.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        operations = [
+            Thread(target=create_files, args=(clients[0], nfs_mount, file_count)),
+            Thread(target=create_hard_link, args=(clients[1], nfs_mount, file_count)),
+            Thread(target=perform_lookups, args=(clients[2], nfs_mount, file_count)),
+        ]
+
+        # start opertaion on each client
+        for operation in operations:
+            operation.start()
+            sleep(0.5)
```
Comment (vamahaja) [inline tests/nfs/nfs_verify_file_ops_hard_links.py:74]: Add comment for explicit sleep

File: tests/nfs/nfs_verify_file_ops_hard_links.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+        # Wait for all operation to finish
+        for operation in operations:
+            operation.join()
+
+    except Exception as e:
+        log.error(f"Error : {e}")
```
Comment (vamahaja) [inline tests/nfs/nfs_verify_file_ops_hard_links.py:81]: Add meaningful error message

File: tests/nfs/nfs_verify_file_ops_hard_links.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    finally:
```
Comment (vamahaja) [inline tests/nfs/nfs_verify_file_ops_hard_links.py:82]: return ???

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: tests/nfs/nfs_verify_file_ops_hard_links.py (hunk 1 above)
Comment (MohitBis) [reply tests/nfs/nfs_verify_file_ops_hard_links.py:75]: Updated

File: tests/nfs/nfs_verify_file_ops_hard_links.py (hunk 2 above)
Comment (MohitBis) [reply tests/nfs/nfs_verify_file_ops_hard_links.py]: Updated

File: tests/nfs/nfs_verify_file_ops_hard_links.py (hunk 3 above)
Comment (MohitBis) [reply tests/nfs/nfs_verify_file_ops_hard_links.py:84]: Updated

**Review by vamahaja — APPROVED** [review]

**Pull request merged**